### PR TITLE
(Docs)(DOC-4702) Missing white space in gen docs

### PR DESCRIPTION
**Pull request by hestonhoffman**
DITA doesn't like HTML line breaks (`<br>`). This causes missing white
space after some of the sentences in our generated docs. It's especially
bad in the tables. To fix, I've turned our generated markdown tables into 
HTML tables and subbed `<br>` with `<p>` tags.

Also: 
- removes the unused bolt_pwsh_reference.md file
- Adds `thead` and `tbody` elements to HTML tables for styling

!no-release-note

### CODE REVIEW

File: documentation/templates/bolt_cmdlet_reference.md.erb [hunk 1]
```diff
@@ -19,9 +19,22 @@ These PowerShell cmdlets are available in the Bolt PowerShell module.
...
+        <th>Description</th>
+        <th align="center">Mandatory?</th>
+    </tr>
+  </thead>
+  <tbody>
+   <% command[:options].each do |option| -%>
+      <tr>
+        <td>`-<%= option[:name] -%><%= option[:arg] %>`</td>
+        <td><%= option[:help_msg].gsub("<", "&lt;").split("\n").join('<p>')%></td>
+       <td align="center"><%= option[:mandatory] ? '✓' : '' %></td>
```
Comment (beechtom) [inline documentation/templates/bolt_cmdlet_reference.md.erb:35]:
```suggestion
        <td align="center"><%= option[:mandatory] ? '✓' : '' %></td>
```

File: documentation/templates/bolt_cmdlet_reference.md.erb [hunk 2]
```diff
@@ -19,9 +19,22 @@ These PowerShell cmdlets are available in the Bolt PowerShell module.
...
+     </tr>
+   <% end %>
+  </tbody>
+</table>
+<% end %>
```
Comment (beechtom) [inline documentation/templates/bolt_cmdlet_reference.md.erb:40]:
```suggestion
<% end %>

```

File: documentation/templates/bolt_cmdlet_reference.md.erb [hunk 3]
```diff
@@ -19,9 +19,22 @@ These PowerShell cmdlets are available in the Bolt PowerShell module.
...
+   <tr>
+        <th>Option</th>
+        <th>Description</th>
+        <th align="center">Mandatory?</th>
+    </tr>
+  </thead>
+  <tbody>
+   <% command[:options].each do |option| -%>
+      <tr>
+        <td>`-<%= option[:name] -%><%= option[:arg] %>`</td>
```
Comment (beechtom) [inline documentation/templates/bolt_cmdlet_reference.md.erb:33]:
```suggestion
        <td>-<%= option[:name] -%><%= option[:arg] %></td>
```

Looks like using HTML tables means the codeblocks won't be generated.

File: documentation/templates/bolt_command_reference.md.erb [hunk 4]
```diff
@@ -26,9 +26,21 @@ These subcommands, actions, and options are available for Bolt.
...
+  <thead>
+    <tr>
+     <th>Option</th>
+     <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% val[:options].each do |option| -%>
+    <tr>
+        <td><% if option[:short] %>`<%= option[:short] %>`, <% end -%>`<%= option[:long] -%><%= option[:arg] %>`</td>
```
Comment (beechtom) [inline documentation/templates/bolt_command_reference.md.erb:39]:
```suggestion
        <td><% if option[:short] %><%= option[:short] %>, <% end -%><%= option[:long] -%><%= option[:arg] %></td>
```

File: documentation/templates/bolt_cmdlet_reference.md.erb (hunk 3 above)
Comment (hestonhoffman) [reply documentation/templates/bolt_cmdlet_reference.md.erb:33]: Oooh, I missed that. We have to replace the backticks with <code>. I'll fix that

File: documentation/templates/bolt_command_reference.md.erb [hunk 5]
```diff
@@ -26,9 +26,21 @@ These subcommands, actions, and options are available for Bolt.
...
+    <tr>
+     <th>Option</th>
+     <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% val[:options].each do |option| -%>
+    <tr>
+        <td><% if option[:short] %><code><%= option[:short] %></code>, <% end -%><code><%= option[:long] -%><%= option[:arg] %></code></td>
+        <td><%= option[:desc].gsub("<br>", "<p>")%></td>
```
Comment (beechtom) [inline documentation/templates/bolt_command_reference.md.erb:40]: We should just join with `<p>` here instead: https://github.com/puppetlabs/bolt/blob/main/rakelib/docs.rake#L59

Comment (hestonhoffman) [reply documentation/templates/bolt_command_reference.md.erb:40]: I wondered where that <br> was coming from 😃

File: documentation/templates/bolt_command_reference.md.erb [hunk 6]
```diff
@@ -26,9 +26,21 @@ These subcommands, actions, and options are available for Bolt.
...
+    </tr>
+  </thead>
+  <tbody>
+    <% val[:options].each do |option| -%>
+    <tr>
+        <td><% if option[:short] %><code><%= option[:short] %></code>, <% end -%><code><%= option[:long] -%><%= option[:arg] %></code></td>
+        <td><%= option[:desc]%></td>
+    </tr>   
+  <% end %>
+  <% end %>
```
Comment (beechtom) [inline documentation/templates/bolt_command_reference.md.erb:43]: This should go after `</table>`

**Review by beechtom — APPROVED** [review]

**Pull request merged**